packages/@aws-cdk/aws-rds/README.md

```diff
@@ -448,3 +450,36 @@ Aurora Serverless Clusters do not support the following features:
 Read more about the [limitations of Aurora Serverless](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless.html#aurora-serverless.limitations)
 
 Learn more about using Amazon Aurora Serverless by reading the [documentation](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless.html)
+
+#### Data API
+
+You can access your Aurora Serverless DB cluster using the built-in Data API. The Data API doesn't require a persistent connection to the DB cluster. Instead, it provides a secure HTTP endpoint and integration with AWS SDKs.
+
+The following example shows granting Data API access to a Lamba function.
+
+```ts
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as rds from '@aws-cdk/aws-rds';
+
+const cluster = new rds.ServerlessCluster(this, 'AnotherCluster', {
+  engine: rds.DatabaseClusterEngine.AURORA_MYSQL,
+  vpc,
+  enableHttpEndpoint: true,
+});
+
+const fn = new lambda.Function(this, 'MyFunction', {
+  runtime: lambda.Runtime.NODEJS_10_X,
+  handler: 'index.handler',
+  code: lambda.Code.fromAsset(path.join(__dirname, 'lambda-handler')),
+  environment: {
+    CLUTSER_ARN: cluster.clusterArn,
+    SECRET_ARN: cluster.secret.secretArn,
+  },
+});
+db.grantDataApi(fn)
+db.secret.grantRead(fn)
```

Comment:
If anything using the Data API needs to read the secret, granting read to the secret within the `grant` method makes sense to me.

Reply:
Yes, anything using the Data API needs to read the secret.

The problem here is that the secret only exists on `ServerlessCluster` but not on `ImportedServerlessCluster`:
https://github.com/aws/aws-cdk/blob/effed09854b6614e75077fd39be8aced69c33582/packages/%40aws-cdk/aws-rds/lib/serverless-cluster.ts#L336-L339

Should we import the secret as well for `ImportedServerlessCluster`?

Otherwise, we have to remove the `grantDataApi` method from the `IServerlessCluster` interface and only include it on `ServerlessCluster` class, leaving the `ImportedServerlessCluster` class with no `grantDataApi` method.

Reply:
Two options that spring to mind:

1. Override the `grant` method on the `ServerlessCluster` to call the `ServerlessClusterBase` `grant`, and then grant read on the secret.

2. Add `public abstract readonly secret?: secretsmanager.ISecret` to `ServerlessClusterBase`, add an optional `secret` to the `ServerlessClusterAttributes`, and then set it on the `ImportedServerlessCluster` if present. Then if there's a secret, use it in the grant method.

I'd be fine with either; the latter would lend support for imported clusters, but it's unclear how useful that would be. We could always do the first now, and the second later. What do you think?

Reply:
Thanks for the feedback @njlynch!

I'm lending towards option 2. This way the `grant` method does the same for both newly created and imported clusters (given the user has specified the secret in the case of imported clusters).